storagemarket/provider.go

```diff
@@ -54,29 +57,65 @@ type Provider struct {
 	fullnodeApi v1api.FullNode
 	//dagStore    stores.DAGStoreWrapper
 
-	transport datatransfer.Transport
+	Transport *datatransfer.MockTransport
 
 	adapter *Adapter
+
+	dealHandlers *dealHandlers
 }
 
-func NewProvider(sqldb *sql.DB, fullnodeApi v1api.FullNode) (*Provider, error) {
-	ctx, cancel := context.WithCancel(context.Background())
+func NewProvider(repoRoot string, dealsDB *db.DealsDB, fullnodeApi v1api.FullNode, addr address.Address) (*Provider, error) {
+	fspath := path.Join(repoRoot, "incoming")
+	err := os.MkdirAll(fspath, os.ModePerm)
+	if err != nil {
+		return nil, err
+	}
+	fs, err := filestore.NewLocalFileStore(filestore.OsPath(fspath))
+	if err != nil {
+		return nil, err
+	}
+
+	newDealPS, err := newDealPubsub()
+	if err != nil {
+		return nil, err
+	}
 
 	return &Provider{
-		ctx:         ctx,
-		cancel:      cancel,
+		// TODO: pass max transfer duration as a param
+		config:      Config{MaxTransferDuration: 30 * time.Second},
+		Address:     addr,
+		newDealPS:   newDealPS,
+		fs:          fs,
 		fullnodeApi: fullnodeApi,
-		db:          db.NewDealsDB(sqldb),
+		db:          dealsDB,
+
+		acceptDealsChan:  make(chan acceptDealReq, 128),
+		failedDealsChan:  make(chan failedDealReq, 128),
+		restartDealsChan: make(chan restartReq, 128),
```

Comment:
@dirkmc @aarshkshah1992 i think this is dangerous - if we have a bug in our code, we won't notice a deadlock, until we hit 128 deals... this channel should be unbuffered IMO as a start... also why 128 and not 1 or 10 or 20?